jbmc/src/java_bytecode/java_object_factory.cpp

```diff
@@ -1172,12 +1118,13 @@ void java_object_factoryt::allocate_nondet_length_array(
   const typet &element_type,
   const source_locationt &location)
 {
-  const auto &length_sym_expr = gen_nondet_int_init(
-    assignments,
-    "nondet_array_length",
+  const auto &length_sym_expr = generate_nondet_int(
     from_integer(0, java_int_type()),
     max_length_expr,
-    location);
+    "nondet_array_length",
+    loc,
```

Comment:
:question: why is `location` replaced with `loc`?

Reply:
Didn't even notice that as CLion suggested `loc` here. `loc` is a field of the object factory and `location` is the parameter to the function. From what I can see these two values are always the same so one of them is unnecessary. For now I changed it back to `location`, and I'll tidy up the `loc`/`location` duplication in a follow-up PR. :tangerine: